src/ledger/tests.zig

```diff
@@ -44,3 +244,175 @@ pub fn freshDir(path: []const u8) !void {
     } else |_| {}
     try std.fs.cwd().makePath(path);
 }
+
+const test_shreds_dir = sig.TEST_DATA_DIR ++ "/shreds";
+
+fn testShreds(comptime filename: []const u8) ![]const Shred {
+    const path = comptimePrint("{s}/{s}", .{ test_shreds_dir, filename });
+    return loadShredsFromFile(std.testing.allocator, path);
+}
+
+/// Read shreds from binary file structured like this:
+/// [shred0_len: u64(little endian)][shred0_payload][shred1_len...
+///
+/// loadShredsFromFile can read shreds produced by saveShredsToFile or this rust function:
+/// ```rust
+/// fn save_shreds_to_file(shreds: &[Shred], path: &str) {
+///     let mut file = std::fs::File::create(path).unwrap();
+///     for shred in shreds {
+///         let payload = shred.payload();
+///         file.write(&payload.len().to_le_bytes()).unwrap();
+///         file.write(payload).unwrap();
+///     }
+/// }
+/// ```
+pub fn loadShredsFromFile(allocator: Allocator, path: []const u8) ![]const Shred {
+    const file = try std.fs.cwd().openFile(path, .{});
+    const reader = file.reader();
+    var shreds = std.ArrayList(Shred).init(allocator);
```

Comment:
Does `errdefer shreds.deinit();` make sense here? Not really an issue though since I believe this is only used in testing anyway

Reply:
c0319bfec52b84410ef3feb0c60c78a1254745c8